specs/botframework-activity/botframework-activity.md

```diff
@@ -176,10 +176,19 @@ The `timestamp` field records the exact UTC time when the activity occurred. Due
 
 `A2043`: Senders SHOULD always use encode the value of `timestamp` fields as UTC, and they SHOULD always include Z as an explicit UTC mark within the value.
 
+### Local timezone
+The `localTimezone` field expresses the timezone where the activity was generated. The value of the `localTimezone` field SHOULD be a time zone name (zone entry) per the IANA Time Zone database. [[15](#References)]
+
+`A2055`: Clients and bots MAY include the `localTimezone` in their activities. If they do so, they SHOULD also include a `localTimestamp` field.
+
+`A2056`: Channels SHOULD preserve `localTimezone` when forwarding activities from a sender to recipient(s).
```

Comment:
@dandriscoll, it depends on whether you adopt my proposal to use ```MUST```. If it's ```MUST```, then processing ageents should reject the value as a protocol violation.

Reply:
I have a practical reason and a principled reason for disagreeing. The practical reason is that the IANA contents are fluid and so any rigorous validation will not be forward-compatible. The principled reason is that this field does not control protocol behavior, so it's not especially important that the sender know that the receiver successfully processed this field.

Any implementation may of course decide to reject unrecognized contents. But for forward-compatibility reasons, I think we should recommend against it.

Reply:
I don't understand what's the difference between a value that controls protocol behavior vs one that does not. Do you apply the same  principle for the value of the ```localTimeStamp```? Will you accept an activity if it contains a ```localTimestamp``` value that doesn't adhere to ISO  8601?

Reply:
In the next draft, I will add a new restriction, if that's the right term, saying that receivers may ignore localTimezones they don't understand.

Reply:
We don't have many behavior-controlling elements within the protocol, but one example is the Payment Request system described here: https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-request-payment?view=azure-bot-service-3.0#process-callbacks. In this part of the protocol, the caller must receive an HTTP 200 to be certain the payment was successfully processed -- if the payment was not processed, it substantially changes what the client does. In this case, the protocol indicates that the service accepted the request, parsed the request, and finished processing the request.

In contrast, for a Message activity (and contents like `localTimezone`), the client gets a confirmation that the request was received and parsed, but not necessarily that everything was successfully understood and processed. If the bot does not understand the `localTimezone` field, it does not need to communicate that through the HTTP response code. Instead, what is more typical, is that the bot will ask the user to clarify via conversation. This is consistent with a user's location that may be in an unexpected format (e.g. lat/lon when street address is required).

Reply:
@dandriscoll... your A2003 and A2004 rules seem to indicate otherwise. It appears that receivers may indeed reject an activity based on the validity of a value. Am I reading those rules incorrectly?

Reply:
A2002 and A2003 are both MAYs. Therefore, the client is not guaranteed an error response. I think we're debating whether a bot MUST generate an error.

The only time we say the bot MUST generate an error is e.g. payments.

Is the *MAY* behavior adequate?

Reply:
I am trying to understand the behavior that you described in your previous response. Obviously, A2002 and A2003 mean that it's possible (the MAY) for a receiver to reject a request based on the semantics of the value. This means that the behavior you described (i.e. sender -> request -> structural compliance -> receiver -> 200 -> sender then subsequent receiver -> request for clarification -> ...) isn't the only possible one. The protocol does provision for the rejection of a request based on semantic validation of the values, even if that's optional. I would have expected such behavior to be a MUST but even with a MAY, the possibility exists. If the possibility exists, i don't see a reason why one couldn't enforce that the value of the localTimezone property MUST adhere to IANA's specification. If there is a new version of IANA's specification, then the BF protocol versioning scheme can accommodate the change.

Reply:
Are we talking about writing the field or reading/validating the field?

If writing the field, `MUST write IANA` is fine.

If reading/validating the field, `MUST generate an error` goes too far.

Reply:
The thread started about the MUST on the value of the property and whether agents SHOULD reject the value. Given A2003 and A2004, I don't think a ```MUST generate an error``` is possible at this stage. We can discuss A2003 and A2004 as part of an evolution :-)